### PR TITLE
Add message key/value size metric in the processor

### PR DESCRIPTION
**Pull request by vahidhashemian**
`processor.message.key.size.bytes`
`processor.message.value.size.bytes`

Current implementation sends two metrics per message. If this could cause stress on the metrics processing side, we could easily change it to send two aggregate metrics per batch of messages.

### CODE REVIEW

File: singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java [hunk 1]
```diff
@@ -549,7 +550,18 @@ private void writeLogMessages(List<LogMessageAndPosition> logMessagesRead)
     }
     List<LogMessage> logMessagesToWrite = Lists.newArrayListWithExpectedSize(numMessages);
     for (LogMessageAndPosition logMessageRead : logMessagesRead) {
-      logMessagesToWrite.add(logMessageRead.getLogMessage());
+      LogMessage logMessage = logMessageRead.getLogMessage();
+      logMessagesToWrite.add(logMessage);
+      OpenTsdbMetricConverter.gauge(
+              SingerMetrics.PROCESSOR_MESSAGE_KEY_SIZE_BYTES,
+              logMessage.getKey().length,
```
Comment (ambud) [inline singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java:557]: This will throw an NPE when key is null

File: singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java [hunk 2]
```diff
@@ -96,7 +99,18 @@ protected int processLogMessageBatch() throws IOException, LogStreamWriterExcept
         // because there is some data to read we need to prepare the commit
         writer.startCommit();
       }
-      writer.writeLogMessageToCommit(message.getLogMessage());
+      LogMessage logMessage = message.getLogMessage();
+      OpenTsdbMetricConverter.gauge(
+              SingerMetrics.PROCESSOR_MESSAGE_KEY_SIZE_BYTES,
+              logMessage.getKey().length,
```
Comment (ambud) [inline singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java:105]: Potential NPE when key is null

Comment (ambud) [reply singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java:105]: We should also add a test to processor to add null keys to capture and validate the null tests.

File: singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java (hunk 1 above)
Comment (vahidhashemian) [reply singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java:557]: Good point. I think we should still send a metric if key is null so we can correlated values for key/value pairs. Would it make sense to send a fixed number like `-1` when key is null?

File: singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java (hunk 2 above)
Comment (vahidhashemian) [reply singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java:105]: Do we want unit test for the metrics piece? Or general unit tests to cover null keys?

Comment (ambud) [reply singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java:105]: General processor unit test to cover null keys

File: singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java (hunk 1 above)
Comment (ambud) [reply singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java:557]: If the key is null the bytes should be 0. -1 will provide an incorrect signal.
